internal/cli/validate/validate_test.go

```diff
@@ -18,22 +16,9 @@ func TestValidation_Run_SmokeTest(t *testing.T) {
 	require.NoError(t, err)
 
 	pathToExamples := "../../../ocf-spec/0.0.1/examples"
-	files, err := ioutil.ReadDir(pathToExamples)
-	require.NoError(t, err)
-
-	var filePaths []string
-	for _, file := range files {
-		if file.IsDir() || !strings.HasSuffix(file.Name(), ".yaml") {
-			continue
-		}
-
-		filePaths = append(filePaths, path.Join(pathToExamples, file.Name()))
-	}
-
-	require.True(t, len(filePaths) > 0)
 
 	// when
-	err = validation.Run(context.Background(), filePaths)
+	err = validation.Run(context.Background(), []string{pathToExamples})
```

Comment:
**stretch:** It would be nice to have a dedicated test for `RecursiveSearch: true/false`. For that, we can use proper `testdata` + a few ideas on how you can do an assertion:
- output log:
    ```go
	    var buff = &bytes.Buffer{}
	    validation, err := validate.New(buff, validate.Options{MaxConcurrency: 5})
    //...
	    // then
	    assert.Contains(t, buff.String(), "Validated 7 files in total")
    ```
- structure `testdata` in a way that invalid manifest is in nested path, so test with `RecursiveSearch: false` passes and with `RecursiveSearch: true` fails as it also visits the invalid manifest.
- have a test hook func in `Run` method that can spy the final list of `filePaths`, e.g. if you will check the `os.OpenFile()` implementation you will see that they also do sth similar.
- don't do the black box testing, but instead test the `v.getFilesToParse(paths)` function separately. - probably the easiest way.
- ...
- `your own idea goes here`

I'm ok with all of above :)

Reply:
Presumably, the most straightforward way of doing this is by using the output log. I thought also about don't do it blackbox but testing the function but that is tricky (you need to export some function that does not really need to be exported or create a separate file for that). Anyway, I chose a solution with the logs and should be fine now.

Reply:
FYI: If you do that as white box testing you don't need to export any function, test will have access to it :) to change it from black box to white box you need to change the package name from `package validate_test` to `package validate`. Of course, it's not always recommended to do that, but in some cases it's allowed :)